hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/HollowScreening.java

```diff
@@ -0,0 +1,180 @@
+            final var key = new JECDSASecp256k1Key(compressedSecp256k1);
+            // replace any JHollowKeys in the req keys that are derived from this active ECDSA key
+            // so that subsequent key activation verifications pass
+            replaceHollowKeysInReqKeysIfNeeded(hollowKeysInOtherReqKeysIndex, evmAddress, key);
+            // check if the evm address is linked to a hollow account and add it to finalizations map
+            addToHollowFinalizationsIfNeeded(evmAddress, key);
+        }
+    }
+
+    public boolean hasReplacedHollowKeysInOtherReqKeys() {
+        return reqOtherPartyKeysDeHollowed != null;
+    }
+
+    public JKey getScreenedPayerKey() {
+        return reqPayerKey;
+    }
+
+    public List<JKey> getOtherPartyKeysDeHollowed() {
+        return reqOtherPartyKeysDeHollowed;
+    }
+
+    public Map<byte[], JKey> getHollowFinalizations() {
+        return hollowFinalizations;
+    }
+
+    private void addToHollowFinalizationsIfNeeded(final byte[] evmAddress, final JECDSASecp256k1Key key) {
+        final var alias = ByteStringUtils.wrapUnsafely(evmAddress);
+        final var aliasedID = AccountID.newBuilder().setAlias(alias).build();
+        final var lookupResult = sigMetaLookup.aliasableAccountSigningMetaFor(aliasedID, linkedRefs);
+        if (!lookupResult.succeeded()) {
+            // in pre-handle context, there may be a hollow account with this evmAddress in the working state, but
+            // `SigMetaDataLookUp`
+            // is not aware of it; or there is a pending hollow account creation txn not yet executed;
+            // in any case, SigMetadataLookup already would have linked the evm address alias and (possibly) account ID
+            // so here link the ECDSA public key alias as well just in case
+            linkPublicKeyAliasIfNeeded(key);
+            return;
+        }
+        final var metadata = lookupResult.metadata();
+        final var accountKey = metadata.key();
+        if (accountKey.hasHollowKey()) {
+            // we have a matched hollow account, so this is a potential finalization;
+            hollowFinalizations.putIfAbsent(evmAddress, key);
+            // in pre-handle context, make sure to update the `linkedRefs` so we handle hollow account’s sig metadata
+            // changes
+            // from another txn until `handle()` of the current one;
+            // SigMetadataLookup already would have linked the evm address alias and (possibly) account ID
+            // so here link the ECDSA public key alias as well just in case
+            linkPublicKeyAliasIfNeeded(key);
+            // DISCUSS we may need to add the JECDSAKey to the req sigs here, if we want to finalize only if the key is
+            // active
+            // otherwise, full-prefix ECDSA sigs would still finalize an account if there is match, even if the sigs are
+            // invalid
```

Comment:
Commenting just to point this out. 

Discussed it briefly with @stoqnkpL, it seems like it would be OK to finalize the accounts if the corresponding `ECDSA` key is found, even if the signature is not verified, since in any case, a subsequent transaction would require a valid signature from the finalized account.